Adding SAMP/T Block 1, Block 1NT and Block 2 (and other changes)

zahnatom opened the pull request:
I was bored
So idk I wrote the code for this, the models and textures were already there

no guarantees for accurate performance though

zahnatom committed:
adding SAMP/T

everything ig

zahnatom committed:
Update SAMPT_MLT_Blk1NT.lua

fix threat range

zahnatom committed:
code for aster 30 block 2 and some fixes

zahnatom committed:
aster 30 block 2 model

zahnatom committed:
aster 30 block 2 textures

zahnatom committed:
forgot the entry.lua, whoops

zahnatom committed:
Update ASTER-30-2.lua

zahnatom committed:
Update ASTER-30-1NT.lua

zahnatom committed:
Update SAMPT_MRI_GF300.lua

zahnatom committed:
fix performance of asters

zahnatom committed:
Update README.md

zahnatom committed:
Update README.md

zahnatom committed:
Update README.md

zahnatom committed:
improvements

fixed launcher animations, adjusted launcher .luas for said changes

zahnatom committed:
further improvements

fixing aster 30 block 1 and sensor height

zahnatom committed:
Update README.md

YoloWingPixie commented on Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1.lua:
```diff
@@ -0,0 +1,96 @@
+	go_active_by_default	= 0,	
+
+	PN_coeffs = {3, 				-- Number of Entries
+				50000.0 ,1.0,		-- Less 5 km to target Pn = 1
```
You have 50km set here, not 5km. I do not think you want to be full lead guidance at 50km

YoloWingPixie commented on Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1.lua:
```diff
@@ -0,0 +1,96 @@
+GT_t.LN_t.Aster30_1.reactionTime = 0.1;
+GT_t.LN_t.Aster30_1.launch_delay = 0.1;
+GT_t.LN_t.Aster30_1.reflection_limit = 0.0001;
+GT_t.LN_t.Aster30_1.ECM_K = 0
```
Again, I believe this should be 0.1 or 0.2, not a flat 0.

YoloWingPixie commented on Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1NT.lua:
```diff
@@ -0,0 +1,96 @@
+    Nr_max 			= 60,
+    v_min 			= 50.0,
+    v_mid 			= 1800.0,
+    Mach_max 		= 5.0,
```
Same question as the Blk1, where does this value come from?

YoloWingPixie commented on Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1NT.lua:
```diff
@@ -0,0 +1,96 @@
+GT_t.LN_t.Aster30_1NT.reactionTime = 0.1;
+GT_t.LN_t.Aster30_1NT.launch_delay = 0.1;
+GT_t.LN_t.Aster30_1NT.reflection_limit = 0.0001;
+GT_t.LN_t.Aster30_1NT.ECM_K = 0
```
Same comment as before, 0.1/0.2 not 0.

YoloWingPixie commented on Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-2.lua:
```diff
@@ -0,0 +1,98 @@
+	go_active_by_default	= 0,	
+
+	PN_coeffs = {3, 				-- Number of Entries
+				100000.0 ,1.0,		-- Less 5 km to target Pn = 1
```
Here we have full lead guidance being implemented at 100km, I do not think this is what you intend. That is a lot of wasted energy if the target manuevers.

YoloWingPixie commented on Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1.lua:
```diff
@@ -0,0 +1,96 @@
+
+    Escort 			= 0,
+    Head_Type 		= 2,
+	sigma 			= {10, 10, 10},
```
Seems kind of low, I think `{20, 20, 20}` would be a little more representative

YoloWingPixie commented on Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-1NT.lua:
Consistent whitespace issues here too

YoloWingPixie commented on Mods/tech/HighDigitSAMs/Database/Weapon/ASTER-30-2.lua:
Consistent whitespace issues

YoloWingPixie reviewed (changes requested):
I've added my comments to this, looks very good. Some small things I'd like to see worked out personally.

zahnatom committed:
improvements

zahnatom committed:
further improvements

zahnatom committed:
Update README.md

update readme to reflect the changes in code

YoloWingPixie commented:
Might have been a display issue, it looks like most of the whitespace is justified now. 

I or @Auranis will test again soon

YoloWingPixie commented:
Yep, we've encountered the same thing with the SA-20 missiles. It was fixable for us by switching over to `PN_Gain` and then cranking it up until the missile barely stays below the altitude limit before it'll escape into space with a worst case target.

zahnatom committed:
fix SA-23 launchers & SA-20 Missiles

the launcher animations and locations of the SA-23 series was bad so i fixed it
SA-20 missiles were affected by ED's TVC change so i updated the TVC value to make them not spin out

YoloWingPixie commented:
Please undo that change, that's already in another pull request

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: zahnatom ***@***.***>
Sent: Friday, June 28, 2024 6:56:25 AM
To: Auranis/HighDigitSAMs ***@***.***>
Cc: Zachary Maynard ***@***.***>; Comment ***@***.***>
Subject: Re: [Auranis/HighDigitSAMs] Adding SAMP/T Block 1, Block 1NT and Block 2 (and other changes) (PR #55)


Decided to fix the SA-23 launchers as well as making the SA-20 missiles not spin out
Hope thats ok

—
Reply to this email directly, view it on GitHub<https://github.com/Auranis/HighDigitSAMs/pull/55#issuecomment-2196984076>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AT5UARINV43W42YLYFIAI3TZJVTQTAVCNFSM6AAAAABI2ERX6SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCOJWHE4DIMBXGY>.
You are receiving this because you commented.Message ID: ***@***.***>

YoloWingPixie commented:
Please undo the changes to the 48N6 and 48N6E2, there's already changes to those missiles in #53

zahnatom committed:
revert changes for 48N6 series

Auranis commented:
Thanks for your implementation @zahnatom we will include this in the upcoming release.
